src/core/Akka.Cluster/SBR/DowningStrategy.cs

```diff
@@ -256,6 +260,51 @@ private ImmutableHashSet<UniqueAddress> AdditionalNodesToDownWhenIndirectlyConne
             }
         }
 
+        #region Unreachable island detection
+        /// <summary>
+        /// Detect floating unreachable islands in the reachability graph.
+        /// 
+        /// Note, this is a potentially expensive operation to perform, this is designed to be called by
+        /// `AdditionalNodesToDownWhenIndirectlyConnected()` which is very rarely called, do not call this anywhere else.
+        /// </summary>
+        /// <param name="reachability">Local copy of the reachability graph</param>
+        /// <param name="unreachable">Local copy of the unreachable addresses</param>
+        /// <returns>Hash set of detected disconnected addresses</returns>
+        private ImmutableHashSet<UniqueAddress> PruneUnreachableNodes(
+            Reachability reachability,
+            ImmutableHashSet<UniqueAddress> unreachable)
+        {
+            // Get all reachable nodes
+            var allReachable = AllMembers.Select(m => m.UniqueAddress)
+                .Where(a => !unreachable.Contains(a))
+                .ToImmutableHashSet();
+                
+            // Collect all nodes that are reported as both record observer and unreachable 
+            // (possible indirect connection)
+            var possibleIndirect = reachability.Records
+                .Where(r => unreachable.Contains(r.Observer))
+                .Select(r => r.Observer)
+                .ToImmutableHashSet(); 
```

Comment:
From all the `Reachability` `Records` that we want to filter, find any `Record.Observer` that is also inside the unreachable set. These are all potential poison records that needs to be cleaned.